serializers/kubernetes_cluster.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class Serializers::KubernetesCluster < Serializers::Base
+  def self.serialize_internal(kc, options = {})
+    base = {
+      id: kc.ubid,
+      name: kc.name,
+      cp_node_count: kc.cp_node_count,
+      private_subnet_id: kc.private_subnet_id,
+      kubernetes_version: kc.kubernetes_version,
+      location: kc.display_location,
+      vms: kc.cp_vms.map { |vm| {name: vm.name, state: vm.display_state, hostname: vm.ephemeral_net4.to_s} },
+      nodepools: kc.kubernetes_nodepools.map { |np| {name: np.name, node_count: np.node_count, vms: np.vms.map { |vm| {name: vm.name, state: vm.display_state, hostname: vm.ephemeral_net4.to_s} }} }
```

Comment:
I highly recommend loading them eagerly; otherwise, you'll end up with a lot of N+1 queries.